shared/src/wg.rs

```diff
@@ -32,17 +32,17 @@ fn cmd(bin: &str, args: &[&str]) -> Result<std::process::Output, io::Error> {
 }
 
 #[cfg(target_os = "macos")]
-pub fn set_addr(interface: &InterfaceName, addr: IpNetwork) -> Result<(), io::Error> {
+pub fn set_addr(interface: &InterfaceName, addr: IpNet) -> Result<(), io::Error> {
     let real_interface = wireguard_control::backends::userspace::resolve_tun(interface)?;
 
-    if addr.is_ipv4() {
+    if matches!(addr, IpNet::V4(_)) {
```

Comment:
Optional: this would be also a candidate for `IpNetExt` convenience method.

Reply:
true!